### ISSUE TITLE
Debugging modules + no such container

### ISSUE THREAD

**Issue by nhuurnink** [Azure/iotedge#59]
Hi there,

I'm trying to debug my module, as I used the sample project from https://docs.microsoft.com/ms-my/azure/iot-edge/tutorial-csharp-module and simply don't get messages in my CSharpSample project.

No errors, running fine, logging that the iot hub module is loaded, and nothing...
I'm running on raspbian (Arm32v7), so I'm not sure how I can debug my modules.

When trying with Arm32v7 as well as Amd64.debug, I get the error: 
Error: No such container: CSharpModule

Any suggestions?

Kind regards, Norbert

**Comment by dsajanice** [comment]
Hi @nhuurnink please refer to the troubleshooting guide (https://docs.microsoft.com/ms-my/azure/iot-edge/troubleshoot) to get some more information on what the issue might be. Do you see messages reaching IoTHub (as documented in https://docs.microsoft.com/ms-my/azure/iot-edge/tutorial-csharp-module#view-generated-data).

**Comment by nhuurnink** [comment]
Well I'm not sure what to do. No I don't see messages reaching IoTHub, at least not using Visual Studio Code and no in Azure Portal "Device to cloud" messages. But there is nothing in the tutorial that mentions any changes on the tempSensor project and I do see tempSensor 'sending messages'  in its log.

Whereas in Visual Studio Code it shows that tempSensor is not running (while it is running on my device, if I run 'iotedge list'  there).

I'm unsure (and confused) about multiple things that happen, which makes it tough.

**Comment by nhuurnink** [comment]
[ERR] - Module My-device-name/CSharpModule is not connected.

is shown is the log for EdgeHub on my device, while iotedge list shows that it's running..

**Comment by dsajanice** [comment]
What logs do you see from the module? 'docker logs CSharpModule'

**Comment by nhuurnink** [comment]
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock

**Comment by nhuurnink** [comment]
The same as when I run iotedge logs CSharpModule. 2 valid logging lines that indicate that the Init() method is run. Also the code:
`    // Attach a callback for updates to the module twin's desired properties.
    await ioTHubModuleClient.SetDesiredPropertyUpdateCallbackAsync(OnDesiredPropertiesUpdate, null);

    // Register a callback for messages that are received by the module.
    await ioTHubModuleClient.SetInputMessageHandlerAsync("input1", FilterMessages, ioTHubModuleClient);`

is in there, but the callbacks are never triggered.

**Comment by dsajanice** [comment]
Could you please run and paste the output of 'sudo docker logs edgeHub | grep INF.*temp | less'. That might give us some clues as to why the temperature module is not sending messages to the cloud. The other thing to check would be the edgeHub routes. The routes can be obtained from the Azure Portal. Please navigate to your Edge device and locate the $edgeHub module. Clicking on it should take you to the 'IoT Edge Module Details' page. Please click on the 'Module Identity Twin' on that page and copy and paste your "routes" here.

**Comment by nhuurnink** [comment]
```
{
  "deviceId": "MI-DataCollector-003-Utrecht-Norbert",
  "moduleId": "$edgeHub",
  "etag": "AAAAAAAAABs=",
  "deviceEtag": "OTY5Njc4NTI3",
  "status": "enabled",
  "statusUpdateTime": "0001-01-01T00:00:00",
  "connectionState": "Disconnected",
  "lastActivityTime": "0001-01-01T00:00:00",
  "cloudToDeviceMessageCount": 0,
  "authenticationType": "sas",
  "x509Thumbprint": {
    "primaryThumbprint": null,
    "secondaryThumbprint": null
  },
  "version": 67,
  "properties": {
    "desired": {
      "schemaVersion": "1.0",
      "routes": {
        "sensorToSampleModule": "FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint(\"/modules/SampleModule/inputs/input1\")",
        "SampleModuleToIoTHub": "FROM /messages/modules/SampleModule/outputs/output1 INTO $upstream"
      },
      "storeAndForwardConfiguration": {
        "timeToLiveSecs": 7200
      },
      "$metadata": {
        "$lastUpdated": "2018-07-23T16:38:18.5823398Z",
        "$lastUpdatedVersion": 27,
        "schemaVersion": {
          "$lastUpdated": "2018-07-23T16:38:18.5823398Z",
          "$lastUpdatedVersion": 27
        },
        "routes": {
          "$lastUpdated": "2018-07-23T16:38:18.5823398Z",
          "$lastUpdatedVersion": 27,
          "sensorToSampleModule": {
            "$lastUpdated": "2018-07-23T16:38:18.5823398Z",
            "$lastUpdatedVersion": 27
          },
          "SampleModuleToIoTHub": {
            "$lastUpdated": "2018-07-23T16:38:18.5823398Z",
            "$lastUpdatedVersion": 27
          }
        },
        "storeAndForwardConfiguration": {
          "$lastUpdated": "2018-07-23T16:38:18.5823398Z",
          "$lastUpdatedVersion": 27,
          "timeToLiveSecs": {
            "$lastUpdated": "2018-07-23T16:38:18.5823398Z",
            "$lastUpdatedVersion": 27
          }
        }
      },
      "$version": 27
    },
    "reported": {
      "schemaVersion": "1.0",
      "version": {
        "version": "1.0.0",
        "build": "14487025",
        "commit": "52ef77db24126bf473265fc09c53d35290a2dd6b"
      },
      "lastDesiredVersion": 27,
      "lastDesiredStatus": {
        "code": 200,
        "description": ""
      },
      "clients": {
        "MI-DataCollector-003-Utrecht-Norbert/tempSensor": {
          "status": "Connected",
          "lastConnectedTimeUtc": "2018-07-23T19:14:12.708532Z"
        },
        "MI-DataCollector-003-Utrecht-Norbert/SampleModule": {
          "status": "Connected",
          "lastConnectedTimeUtc": "2018-07-23T19:17:41.3480388Z"
        }
      },
      "$metadata": {
        "$lastUpdated": "2018-07-23T19:17:41.3806611Z",
        "schemaVersion": {
          "$lastUpdated": "2018-07-23T19:17:41.3806611Z"
        },
        "version": {
          "$lastUpdated": "2018-07-23T19:17:41.3806611Z",
          "version": {
            "$lastUpdated": "2018-07-23T19:17:41.3806611Z"
          },
          "build": {
            "$lastUpdated": "2018-07-23T19:17:41.3806611Z"
          },
          "commit": {
            "$lastUpdated": "2018-07-23T19:17:41.3806611Z"
          }
        },
        "lastDesiredVersion": {
          "$lastUpdated": "2018-07-23T19:13:50.6191209Z"
        },
        "lastDesiredStatus": {
          "$lastUpdated": "2018-07-23T19:13:50.6191209Z",
          "code": {
            "$lastUpdated": "2018-07-23T19:13:50.6191209Z"
          },
          "description": {
            "$lastUpdated": "2018-07-23T19:13:50.6191209Z"
          }
        },
        "clients": {
          "$lastUpdated": "2018-07-23T19:17:41.3806611Z",
          "MI-DataCollector-003-Utrecht-Norbert/tempSensor": {
            "$lastUpdated": "2018-07-23T19:14:12.7620624Z",
            "status": {
              "$lastUpdated": "2018-07-23T19:14:12.7620624Z"
            },
            "lastConnectedTimeUtc": {
              "$lastUpdated": "2018-07-23T19:14:12.7620624Z"
            }
          },
          "MI-DataCollector-003-Utrecht-Norbert/SampleModule": {
            "$lastUpdated": "2018-07-23T19:17:41.3806611Z",
            "status": {
              "$lastUpdated": "2018-07-23T19:17:41.3806611Z"
            },
            "lastConnectedTimeUtc": {
              "$lastUpdated": "2018-07-23T19:17:41.3806611Z"
            }
          }
        }
      },
      "$version": 40
    }
  }
}
```

**Comment by nhuurnink** [comment]
2018-07-23 12:43:32.502 +00:00 [INF] - Attempting to connect to IoT Hub for client MI-DataCollector-003-Utrecht-Norbert/$edgeHub via AMQP...
2018-07-23 12:43:59.815 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 12:44:18.175 +00:00 [INF] - Attempting to connect to IoT Hub for client MI-DataCollector-003-Utrecht-Norbert/tempSensor via AMQP...
2018-07-23 12:44:18.201 +00:00 [INF] - Connected to IoT Hub for client MI-DataCollector-003-Utrecht-Norbert/tempSensor via AMQP, with client operation timeo                                                                                 ut 60000.
2018-07-23 12:44:18.319 +00:00 [INF] - Created cloud connection for client MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 12:44:18.320 +00:00 [INF] - New cloud connection created for device MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 12:44:18.488 +00:00 [INF] - New token requested by client MI-DataCollector-003-Utrecht-Norbert/tempSensor, but using existing token as it is usab                                                                                 le.
2018-07-23 12:44:19.946 +00:00 [INF] - New device connection for device MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 12:44:20.148 +00:00 [INF] - Bind device proxy for device MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 12:44:20.159 +00:00 [INF] - Initialized AMQP connection handler for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 12:44:20.283 +00:00 [INF] - Opened link Events for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 12:44:21.166 +00:00 [INF] - Opened link MethodReceiving for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 12:44:21.946 +00:00 [INF] - Opened link MethodSending for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 12:44:22.382 +00:00 [INF] - Opened link ModuleMessages for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 12:46:19.940 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 13:03:10.313 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 13:18:44.097 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 13:24:42.308 +00:00 [INF] - Closing link Events for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 13:24:42.314 +00:00 [INF] - Closing link MethodReceiving for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 13:24:42.330 +00:00 [INF] - Closing link MethodSending for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 13:24:42.330 +00:00 [INF] - Closing link ModuleMessages for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 13:26:39.143 +00:00 [INF] - Attempting to connect to IoT Hub for client MI-DataCollector-003-Utrecht-Norbert/$edgeHub via AMQP...
2018-07-23 13:26:59.538 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 13:27:15.880 +00:00 [INF] - Attempting to connect to IoT Hub for client MI-DataCollector-003-Utrecht-Norbert/tempSensor via AMQP...
2018-07-23 13:27:15.892 +00:00 [INF] - Connected to IoT Hub for client MI-DataCollector-003-Utrecht-Norbert/tempSensor via AMQP, with client operation timeo                                                                                 ut 60000.
2018-07-23 13:27:15.894 +00:00 [INF] - Created cloud connection for client MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 13:27:15.894 +00:00 [INF] - New cloud connection created for device MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 13:27:15.968 +00:00 [INF] - New token requested by client MI-DataCollector-003-Utrecht-Norbert/tempSensor, but using existing token as it is usab                                                                                 le.
2018-07-23 13:27:17.082 +00:00 [INF] - New device connection for device MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 13:27:17.188 +00:00 [INF] - Bind device proxy for device MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 13:27:17.198 +00:00 [INF] - Initialized AMQP connection handler for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 13:27:17.313 +00:00 [INF] - Opened link Events for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 13:27:18.210 +00:00 [INF] - Opened link MethodReceiving for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 13:27:18.856 +00:00 [INF] - Opened link MethodSending for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 13:27:19.161 +00:00 [INF] - Opened link ModuleMessages for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 13:44:02.689 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 13:45:52.899 +00:00 [INF] - Attempting to connect to IoT Hub for client MI-DataCollector-003-Utrecht-Norbert/SampleModule via AMQP...
2018-07-23 14:11:56.102 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 14:14:53.732 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 14:15:54.024 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 14:23:05.710 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 14:23:45.767 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 14:25:14.013 +00:00 [INF] - Existing token not found for MI-DataCollector-003-Utrecht-Norbert/tempSensor. Getting new token from the client...
2018-07-23 14:25:14.299 +00:00 [INF] - Obtained new token for client MI-DataCollector-003-Utrecht-Norbert/tempSensor that expires in 00:01:59.7008361
2018-07-23 14:27:05.101 +00:00 [INF] - Existing token not found for MI-DataCollector-003-Utrecht-Norbert/tempSensor. Getting new token from the client...
2018-07-23 14:27:05.103 +00:00 [INF] - Obtained new token for client MI-DataCollector-003-Utrecht-Norbert/tempSensor that expires in 00:58:08.8965677
2018-07-23 14:27:45.730 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 14:28:00.824 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 14:28:43.087 +00:00 [INF] - Attempting to connect to IoT Hub for client MI-DataCollector-003-Utrecht-Norbert/SampleModule via AMQP...
2018-07-23 14:41:03.905 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 14:41:48.034 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 14:44:37.975 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 14:48:09.598 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 14:48:30.174 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 14:48:47.836 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 14:49:40.646 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 14:50:10.292 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 16:27:59.651 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 16:29:43.515 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 16:33:23.299 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 16:37:43.971 +00:00 [INF] - Attempting to connect to IoT Hub for client MI-DataCollector-003-Utrecht-Norbert/SampleModule via AMQP...
2018-07-23 16:38:19.337 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/Sam                                                                                 pleModule/inputs/input1")
2018-07-23 18:39:59.368 +00:00 [INF] - Closing link Events for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 18:39:59.374 +00:00 [INF] - Closing link MethodReceiving for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 18:39:59.375 +00:00 [INF] - Closing link MethodSending for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 18:39:59.376 +00:00 [INF] - Closing link ModuleMessages for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 19:13:29.930 +00:00 [INF] - Attempting to connect to IoT Hub for client MI-DataCollector-003-Utrecht-Norbert/$edgeHub via AMQP...
2018-07-23 19:13:52.250 +00:00 [INF] - sensorToSampleModule: FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint("/modules/SampleModule/inputs/input1")
2018-07-23 19:14:11.120 +00:00 [INF] - Attempting to connect to IoT Hub for client MI-DataCollector-003-Utrecht-Norbert/tempSensor via AMQP...
2018-07-23 19:14:11.146 +00:00 [INF] - Connected to IoT Hub for client MI-DataCollector-003-Utrecht-Norbert/tempSensor via AMQP, with client operation timeout 60000.
2018-07-23 19:14:11.150 +00:00 [INF] - Created cloud connection for client MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 19:14:11.150 +00:00 [INF] - New cloud connection created for device MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 19:14:11.313 +00:00 [INF] - New token requested by client MI-DataCollector-003-Utrecht-Norbert/tempSensor, but using existing token as it is usable.
2018-07-23 19:14:12.669 +00:00 [INF] - New device connection for device MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 19:14:12.769 +00:00 [INF] - Bind device proxy for device MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 19:14:12.779 +00:00 [INF] - Initialized AMQP connection handler for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 19:14:12.895 +00:00 [INF] - Opened link Events for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 19:14:13.715 +00:00 [INF] - Opened link MethodReceiving for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 19:14:14.362 +00:00 [INF] - Opened link MethodSending for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 19:14:14.722 +00:00 [INF] - Opened link ModuleMessages for MI-DataCollector-003-Utrecht-Norbert/tempSensor
2018-07-23 19:17:40.798 +00:00 [INF] - Attempting to connect to IoT Hub for client MI-DataCollector-003-Utrecht-Norbert/SampleModule via AMQP...

**Comment by nhuurnink** [comment]
What surprises me is the connection attempt via AMQP, as the SampleModule (CSharpModule name in my example) is trying to send messages through MQTT (as transporttype in code)

**Comment by dsajanice** [comment]
Looks like you have found a bug in our documentation. The https://docs.microsoft.com/ms-my/azure/iot-edge/tutorial-csharp-module#build-your-iot-edge-solution step in the doc is missing a step to change the routes to reflect the name of the module. Please run through that step again and rename 'SampleModule' in the deployment.template.json to 'CSharpModule'.

Regarding your second comment about protocol used. The EdgeHub is using AMQP to connect to IoTHub and the CSharpModule is using MQTT to connect to EdgeHub. This is a valid and supported scenario.

**Comment by nhuurnink** [comment]
Wait, I used the name SampleModule everywhere, but to make it clear that I was following the tutorial, I change it to CSharpModule in my question. Until the copy-paste actions that is, because that would be alot of work.

So the name of my module is SampleModule.

**Comment by nhuurnink** [comment]
```
    "$edgeHub": {
      "properties.desired": {
        "schemaVersion": "1.0",
        "routes": {
          "sensorToSampleModule": "FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint(\"/modules/SampleModule/inputs/input1\")",
          "SampleModuleToIoTHub": "FROM /messages/modules/SampleModule/outputs/output1 INTO $upstream"
        },
        "storeAndForwardConfiguration": {
          "timeToLiveSecs": 7200
        }
      }
    },
    "SampleModule": {
      "properties.desired":{
          "TemperatureThreshold":10
      }
```

**Comment by dsajanice** [comment]
@shizn Could you please take a look?

**Comment by dsajanice** [comment]
Could you please paste the output of 'journalctl -u iotedge.service -b | grep SampleModule'

**Comment by nhuurnink** [comment]
Jul 24 06:10:51 raspberrypi iotedged[508]: 2018-07-24T06:10:51Z [INFO] - [mgmt] - - - [2018-07-24 06:10:51.852362785 UTC] "PUT /identities SampleModule?api-version=2018-06-28 HTTP/1.1" 200 OK 102 "-" "-" pid(800)
Jul 24 06:10:53 raspberrypi iotedged[508]: 2018-07-24T06:10:53Z [INFO] - [mgmt] - - - [2018-07-24 06:10:53.676703358 UTC] "POST /modules/SampleModule/stop?api-version=2018-06-28 HTTP/1.1" 304 Not Modified - "-" "-" pid(800)
Jul 24 06:10:54 raspberrypi iotedged[508]: 2018-07-24T06:10:54Z [INFO] - Updating module SampleModule
Jul 24 06:10:57 raspberrypi iotedged[508]: 2018-07-24T06:10:57Z [INFO] - [mgmt] - - - [2018-07-24 06:10:57.679206377 UTC] "PUT /modules/SampleModule?api-version=2018-06-28 HTTP/1.1" 200 OK 1035 "-" "-" pid(800)
Jul 24 06:11:00 raspberrypi iotedged[508]: 2018-07-24T06:11:00Z [INFO] - [mgmt] - - - [2018-07-24 06:11:00.082746793 UTC] "POST /modules/SampleModule/start?api-version=2018-06-28 HTTP/1.1" 204 No Content - "-" "-" pid(800)
Jul 24 06:15:24 raspberrypi iotedged[508]: 2018-07-24T06:15:24Z [INFO] - [mgmt] - - - [2018-07-24 06:15:24.337824004 UTC] "POST /modules/SampleModule/stop?api-version=2018-06-28 HTTP/1.1" 304 Not Modified - "-" "-" pid(800)
Jul 24 06:15:26 raspberrypi iotedged[508]: 2018-07-24T06:15:26Z [INFO] - [mgmt] - - - [2018-07-24 06:15:26.633417595 UTC] "POST /modules/SampleModule/start?api-version=2018-06-28 HTTP/1.1" 204 No Content - "-" "-" pid(800)
Jul 24 06:15:31 raspberrypi iotedged[508]: 2018-07-24T06:15:31Z [INFO] - [work] - - - [2018-07-24 06:15:31.962452210 UTC] "POST /modules/SampleModule/genid/636679541274465391/sign?api-version=2018-06-28 HTTP/1.1" 200 OK 57 "-" "-" pid(2480)

**Comment by dsajanice** [comment]
I was able to get this setup to work by changing the transport type in SampleModule from MQTT to AMQP as follows:
`//MqttTransportSettings mqttSetting = new MqttTransportSettings(TransportType.Mqtt_Tcp_Only);
//ITransportSettings[] settings = { mqttSetting };`

`AmqpTransportSettings amqpSetting = new AmqpTransportSettings(TransportType.Amqp_Tcp_Only);
ITransportSettings[] settings = {amqpSetting};
`
Could you please try this and let me know if it works.

**Comment by nhuurnink** [comment]
Yes now it works with AMQP. Which makes somewhat sense, since we saw in the logs earlier: Attempting to connect to IoT Hub for client MI-DataCollector-003-Utrecht-Norbert/SampleModule via AMQP...

It seems wrong that it's doing that after we create the iotHubModuleClient with MQTP_tcp_only settings.

I really do want to use MQTT for my usecase :)

**Comment by dsajanice** [comment]
The log line 'Attempting to connect to IoT Hub for client MI-DataCollector-003-Utrecht-Norbert/SampleModule via AMQP...' refers to the connection that Edge Hub is trying to establish with the cloud. It is independent of the connection that the module/leaf device is establishing with Edge Hub. As I mentioned earlier, a module/leaf device should be able to connect to Edge Hub using MQTT or AMQP and Edge Hub can connect to the cloud with either AMQP or MQTT. The two protocols are independent of each other.

I have filed a bug for the MQTT issue.

**Comment by myagley** [comment]
The MQTT issue described is now fixed with version 1.18.0 of the C# SDK. Closing this issue for now, but please feel free to reopen if there are any questions/concerns.
